index.js

```diff
@@ -170,10 +170,11 @@ function serveIndex(root, options) {
         fstat(path, files, function (err, stats) {
           if (err) return next(err);
 
-          // combine the stats into the file list
+          // combine the stats into the file list,
+          // ignoring ENOENT / null stat objects
           var fileList = files.map(function (file, i) {
             return { name: file, stat: stats[i] };
-          });
+          }).filter(function (file) { return file.stat });
```

Comment:
Will this actually be filtering anything out since the code below is providing a stat object even on error?